pubsub/google/cloud/pubsub/topic.py

```diff
@@ -464,16 +480,28 @@ def publish(self, message, **attrs):
         :param attrs: key-value pairs to send as message attributes
         """
         self.topic._timestamp_message(attrs)
-        self.messages.append(
-            {'data': message,
-             'attributes': attrs})
+
+        # Append the message to the list of messages..
+        item = {'attributes': attrs, 'data': message}
+        self.messages.append(item)
+
+        # Determine the approximate size of the message, and increment
+        # the current batch size appropriately.
+        encoded = b''
+        if isinstance(message, six.text_type):
+            encoded += base64.b64encode(message.encode('utf8'))
+        else:
+            encoded += base64.b64encode(message)
+        encoded += base64.b64encode(
+            json.dumps(attrs, ensure_ascii=False).encode('utf8'),
```

Comment:
Additionally, it _probably_ does not matter if we lost some characters at this point. What matters is that the size is "close enough".